src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -492,47 +497,285 @@ antlrcpp::Any CypherMainVisitor::visitStreamQuery(MemgraphCypher::StreamQueryCon
 }
 
 antlrcpp::Any CypherMainVisitor::visitCreateStream(MemgraphCypher::CreateStreamContext *ctx) {
+  MG_ASSERT(ctx->children.size() == 1, "CreateStreamQuery should have exactly one child!");
+  auto *stream_query = ctx->children[0]->accept(this).as<StreamQuery *>();
+  query_ = stream_query;
+  return stream_query;
+}
+
+namespace {
+std::vector<std::string> TopicNamesFromSymbols(
+    antlr4::tree::ParseTreeVisitor &visitor,
+    const std::vector<MemgraphCypher::SymbolicNameWithDotsAndMinusContext *> &topic_name_symbols) {
+  MG_ASSERT(!topic_name_symbols.empty());
+  std::vector<std::string> topic_names;
+  topic_names.reserve(topic_name_symbols.size());
+  std::transform(topic_name_symbols.begin(), topic_name_symbols.end(), std::back_inserter(topic_names),
+                 [&visitor](auto *topic_name) { return JoinSymbolicNamesWithDotsAndMinus(visitor, *topic_name); });
+  return topic_names;
+}
+
+template <typename T>
+concept EnumUint8 = std::is_enum_v<T> && std::same_as<uint8_t, std::underlying_type_t<T>>;
```

Comment:
That is true and it actually works like that!
I'm still thinking of keeping `std::is_enum_v<T>` to have a better error message. Without you get:
```
because substituted constraint expression is ill-formed: no type named 'type' in 'std::underlying_type<std::basic_string<char>>'
```

Reply:
About the error message, there are two perspectives on this. There is always the argument that an error message not clearly describing the issue is not a good practice and not explicit. However, the other perspective says: `a constraint just failed` what else do you need and why do you even need an error message since you already know explicitly which constraint failed (so the information is actually there)? I personally subscribe to the second because as long as I don't have 100 lines of failed template instantiations I am happy :D Same for assert, why do you need an error message when you know the condition an assert failed on (these are Walter's E Brown's words -- he brought assert without error messages for exactly this reason). All in all, since you want to opt in for a more explicit error message that's fine by me but I thought it's worthwhile discussing about the two :D

Reply:
I agree, and it really does make sense. :smile: 
But still, concepts can consist of multiple checks and you want to know why did it fail. For example, check out the Stream concept, it contains a bunch of things and you would really like to know which constraint failed.
This thing also, you know a constraint failed, great! But you need to fix it so you still need to spend time figuring out why did it fail. You can argue that the concept, in this case, is simple so both approaches would still work.
All in all, I see your point and thanks for raising it up (and writing such a detailed response) :smile: